services/pdf.py

```
@@ -0,0 +1,64 @@
+
+def save_pdf_for_user(
+    pnr: str,
+    file_storage,
+    categories: Sequence[str],
+    logger: logging.Logger | None = None,
+):
+    """Validera och spara PDF i databasen för angivet personnummer."""
+    logger = logger or logging.getLogger(__name__)
+    if file_storage.filename == "":
+        logger.error("No file selected for upload")
+        raise ValueError("Ingen fil vald.")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)